commons/src/main/java/com/powsybl/commons/extensions/ExtensionJsonSerializer.java

```diff
@@ -23,4 +23,8 @@
     void serialize(E extension, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException;
 
     E deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException;
+
+    default E deserializeAndUpdate(JsonParser jsonParser, DeserializationContext deserializationContext, E parameters) throws IOException {
+        return deserialize(jsonParser, deserializationContext);
```

Comment:
You're right that it's error-prone. I think the options are:
* do it this way, and existing extensions will keep behaving as today, BUT new extensions could "forget" to actually implement this
* throw an exception as default implementation, but I think it's not a good option since it will compile but fail at runtime
* do not implement it, and make it a breaking change, but in that case it's gonna be inconvenient for users

Maybe the best option is to keep the current approach but insist in javadoc that the method must be implemented for updates to work correctly ?

Reply:
I added some javadoc to insist on the need to implement the method for the feature to work properly.

Reply:
I agree it should also be documented in the migration guide for people/projects with custom extensions in my opinion.

Reply:
I have added a section in the migration guide !